src/signal_processing_algorithms/homogeneity.py

```diff
@@ -0,0 +1,93 @@
+"""Homogeneity Calculator."""
+from typing import Callable, List, Union
+
+import numpy as np
+
+from signal_processing_algorithms import difference
+
+
+class HomogeneityCalculator:
+    """A calculator for determining the homogeneity of two distributions."""
+
+    def __init__(self, avg: Callable[..., float] = np.mean) -> None:  # type: ignore
+        """
+        Initialize.
+
+        :param avg: Function to calculate average.
+        """
+        self._avg = avg
+
+    @classmethod
+    def get_valid_type(cls, series: Union[List[float], np.ndarray]) -> np.ndarray:
+        """
+        Return a valid series type.
+
+        :param series: Distribution.
+        :return: A 1D numpy array of floats.
+        """
+        if series is None or len(series) == 0:
+            raise ValueError("Distribution cannot be empty.")
+
+        if not isinstance(series, np.ndarray):
+            series = np.array(series, dtype=np.float64)
+
+        if series.dtype is not np.float64:
+            series = np.array(series, dtype=np.float64)
+
+        return series
+
+    def _get_energy_coefficient_of_inhomogeneity(
+        self,
+        pairwise_distance_in_x: np.ndarray,
+        pairwise_distance_in_y: np.ndarray,
+        pairwise_distance_in_x_y: np.ndarray,
+    ) -> float:
+        """
+        Return Energy coefficient of inhomogeneity from distance matrices.
+
+        :param pairwise_distance_in_x: Distance matrix for pairwise distance in the distribution x.
+        :param pairwise_distance_in_y: Distance matrix for pairwise distance in the distribution y.
+        :return: Energy coefficient of inhomogeneity.
+        """
+        expectation_x = self._avg(pairwise_distance_in_x)
+        expectation_y = self._avg(pairwise_distance_in_y)
+        expectation_x_y = self._avg(pairwise_distance_in_x_y)
+
+        common_term = 2 * expectation_x_y
+        inhomogeneity_coefficient = (common_term - expectation_x - expectation_y) / common_term
+        assert 0 <= inhomogeneity_coefficient <= 1
```

Comment:
Assert in source code? I think you want to raise your own error, or do an assert in a test instead. Probably the latter since this just checks that your calculation worked.